### DISCUSSION TITLE
Node Bypass Input

### DISCUSSION THREAD

**Discussion by VikingTechLLC** [comfyanonymous/ComfyUI#1458] [Ideas]
Possibility of including a "bypass input"?  

Instead of having "on/off" switches, would it be possible to have an additional input on nodes (or groups somehow), where a boolean input would control whether a node/group gets put into bypass mode?

**Comment by chrisgoringe** [comment]
You can right-click on a node and select 'Bypass' already - and I think the same is true of groups as well. Not sure what you're asking for beyond that?

**Reply by VikingTechLLC** [reply]
A way to automate the process.  Say I have a boolean operated switch which activates a block of controlnet nodes, that boolean would also unbypass the nodes or grouping they are organized in.

**Comment by wcurti** [comment]
This would also help for API users

**Reply by digitaljohn** [reply]
100% It would be great to have this for versatile workflows. Is this some it would be worth me looking to implement and may be merged in?

**Reply by ryanontheinside** [reply]
yes

**Reply by tsogzark** [reply]
You can set all node to be always mode and then delete and relink nodes by their title. There is a simple function.
```python
def bypass_node_by_title(nodes: dict, start_with: str):
    keys_delete = []
    for key, node in nodes.items():
        title: str = node["_meta"]["title"]
        inputs: dict = node["inputs"]
        if title.startswith(start_with):
            for _key, _node in nodes.items():
                _node_inputs: dict = _node["inputs"]
                for __key, __node in _node_inputs.items():
                    if isinstance(__node, list):
                        if __node[0] == key:
                            if __key in inputs:
                                _node_inputs[__key] = inputs[__key]
            keys_delete.append(key)
    for key in keys_delete:
        del nodes[key]
```

**Reply by alonsojr1980** [reply]
It would be great if you included how and where to call that function. Thanks.

**Reply by tsogzark** [reply]
Sorry for reply so late. I can't share all code. And the following is the principle how I build a dynamic API based on ComfyUI.
1. Build the large workflow which contains all of your sub workflows and set all nodes to be always mode.
2. Specify a prefix of nodes for your sub workflow, such as inpaint_sampler, inpaint_vae, controlnet_sampler...
3. Export the workflow json by saving as API format.
4. Use json.load to load the workflow as nodes params in this function.
5. Bypass the sub workflow by their title prefix

**Comment by bxclib2** [comment]
Hi, bypass in api is super useful. Any progress for the api implementation?

**Comment by Lumoria** [comment]
Is there still no way to bypass a node through the API?

**Comment by btaens** [comment]
This would indeed be very useful in the API.

**Comment by edtjulien** [comment]
I confirm. It could be great to have this on API.
